dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKFunctionMetadataExtensions.cs

```diff
@@ -12,34 +13,51 @@ public static class SKFunctionMetadataExtensions
     /// <summary>
     /// Convert a <see cref="SKFunctionMetadata"/> to an <see cref="OpenAIFunction"/>.
     /// </summary>
-    /// <param name="functionView">The <see cref="SKFunctionMetadata"/> object to convert.</param>
+    /// <param name="metadata">The <see cref="SKFunctionMetadata"/> object to convert.</param>
     /// <returns>An <see cref="OpenAIFunction"/> object.</returns>
-    public static OpenAIFunction ToOpenAIFunction(this SKFunctionMetadata functionView)
+    public static OpenAIFunction ToOpenAIFunction(this SKFunctionMetadata metadata)
     {
         var openAIParams = new List<OpenAIFunctionParameter>();
-        foreach (SKParameterMetadata param in functionView.Parameters)
+        foreach (SKParameterMetadata param in metadata.Parameters)
         {
+            // Get the parameter's schema, or if it doesn't have one but it has a .NET type,
+            // infer the schema from the .NET type.
+            SKJsonSchema? schema = param.Schema ?? OpenAIFunction.GetJsonSchema(param.ParameterType, param.Description);
+
+            // Read the "type" property from the schema, if it exists.
+            // If it doesn't, that means we lack any type information for the property.
+            string? type = null;
+            if (schema?.RootElement.TryGetProperty("type", out JsonElement prop) == true)
+            {
+                type = prop.GetString();
+            }
+            type ??= "string";
```

Comment:
Is `type` used for anything?

Reply:
Hah, it was, and then I subsequently deleted OpenAIFunctionParameter.Type, at which point this is dead.